src/spikeinterface/core/zarrextractors.py

```diff
@@ -0,0 +1,558 @@
+    folder_path: str or Path
+        Path to the zarr root file
+    storage_options: dict or None
+        Storage options for zarr `store`. E.g., if "s3://" or "gcs://" they can provide authentication methods, etc.
+
+    Returns
+    -------
+    extractor: ZarrExtractor
+        The loaded extractor
+    """
+    root = zarr.open(str(folder_path), mode="r", storage_options=storage_options)
+    if "channel_ids" in root.keys():
+        return read_zarr_recording(folder_path, storage_options=storage_options)
+    elif "unit_ids" in root.keys():
+        return read_zarr_sorting(folder_path, storage_options=storage_options)
+    else:
+        raise ValueError("Cannot find 'channel_ids' or 'unit_ids' in zarr root. Not a valid SpikeInterface zarr format")
+
+
+### UTILITY FUNCTIONS ###
+def resolve_zarr_path(folder_path: str | Path):
+    """
+    Resolve a path to a zarr folder.
+
+    Parameters
+    ----------
+    """
+    if str(folder_path).startswith("s3:") or str(folder_path).startswith("gcs:"):
+        # cloud location, no need to resolve
+        return folder_path, folder_path
+    else:
+        folder_path = Path(folder_path)
+        folder_path_kwarg = str(Path(folder_path).resolve())
+        return folder_path, folder_path_kwarg
+
+
+def get_default_zarr_compressor(clevel: int = 5):
+    """
+    Return default Zarr compressor object for good preformance in int16
+    electrophysiology data.
+
+    cname: zstd (zstandard)
+    clevel: 5
+    shuffle: BITSHUFFLE
+
+    Parameters
+    ----------
+    clevel : int, default: 5
+        Compression level (higher -> more compressed).
+        Minimum 1, maximum 9. By default 5
+
+    Returns
+    -------
+    Blosc.compressor
+        The compressor object that can be used with the save to zarr function
+    """
+    assert ZarrRecordingExtractor.installed, ZarrRecordingExtractor.installation_mesg
+    from numcodecs import Blosc
+
+    return Blosc(cname="zstd", clevel=clevel, shuffle=Blosc.BITSHUFFLE)
+
+
+def add_properties_and_annotations(
+    zarr_group: zarr.hierarchy.Group, recording_or_sorting: Union[BaseRecording, BaseSorting]
+):
+    # save properties
+    prop_group = zarr_group.create_group("properties")
+    for key in recording_or_sorting.get_property_keys():
+        values = recording_or_sorting.get_property(key)
+        prop_group.create_dataset(name=key, data=values, compressor=None)
+
+    # save annotations
+    zarr_group.attrs["annotations"] = check_json(recording_or_sorting._annotations)
+
+
+def add_sorting_to_zarr_group(sorting: BaseSorting, zarr_group: zarr.hierarchy.Group, **kwargs):
+    """
+    Add a sorting extractor to a zarr group.
+
+    Parameters
+    ----------
+    sorting: BaseSorting
+        The sorting extractor object to be added to the zarr group
+    zarr_group: zarr.hierarchy.Group
+        The zarr group
+    kwargs: dict
+        Other arguments passed to the zarr compressor
+    """
+    from numcodecs import Delta
```

Comment:
Does this delta encoding will be a definitive choice ?
I guess that it has better compression no ?
But if one day we want to change this choice then we have a different version of our format.
For instance does this delta is also good for spike unit_index ?

Reply:
it's best for spike samples because it just takes the diff between consecutive samples before standard compression.